pkg/plugins/policies/meshhttproute/api/v1alpha1/compare_test.go

```diff
@@ -0,0 +1,156 @@
+			Matches: matches,
+		})
+	}
+
+	return rules
+}
+
+func makeRoutes(matches []api.Match) []api.Route {
+	var routes []api.Route
+	for _, match := range matches {
+		routes = append(routes, api.Route{
+			Hash:  api.HashMatches([]api.Match{match}),
+			Match: match,
+		})
+	}
+
+	return routes
+}
+
+var _ = Describe("SortRules", func() {
+	exactMatch := api.Match{
+		Path: &api.PathMatch{
+			Type:  api.Exact,
+			Value: "/exact",
+		},
+	}
+	otherExactMatch := api.Match{
+		Path: &api.PathMatch{
+			Type:  api.Exact,
+			Value: "/other-exact",
+		},
+	}
+	prefixMatch := api.Match{
+		Path: &api.PathMatch{
+			Type:  api.PathPrefix,
+			Value: "/prefix",
+		},
+	}
+	longerPrefixMatch := api.Match{
+		Path: &api.PathMatch{
+			Type:  api.PathPrefix,
+			Value: "/prefix/plusmore",
+		},
+	}
+	regexMatch := api.Match{
+		Path: &api.PathMatch{
+			Type:  api.RegularExpression,
+			Value: "/exact.*",
+		},
+	}
+	methodMatch := api.Match{
+		Method: pointer.To(api.Method("GET")),
+	}
+	singleHeaderMatch := api.Match{
+		Headers: []common_api.HeaderMatch{{
+			Type:  pointer.To(common_api.HeaderMatchExact),
+			Name:  "header",
+			Value: "value",
+		}},
+	}
+	It("handles path matches", func() {
+		Expect(api.SortRules(makeSingleMatchRules([]api.Match{
+			prefixMatch,
+			longerPrefixMatch,
+			regexMatch,
+			exactMatch,
+		}))).To(Equal(makeRoutes([]api.Match{
+			exactMatch,
+			longerPrefixMatch,
+			prefixMatch,
+			regexMatch,
+		})))
+	})
+	It("handles different kinds matches", func() {
+		Expect(api.SortRules(makeSingleMatchRules([]api.Match{
+			methodMatch,
+			exactMatch,
+		}))).To(Equal(makeRoutes([]api.Match{
+			exactMatch,
+			methodMatch,
+		})))
+		Expect(api.SortRules(makeSingleMatchRules([]api.Match{
+			singleHeaderMatch,
+			exactMatch,
+		}))).To(Equal(makeRoutes([]api.Match{
+			exactMatch,
+			singleHeaderMatch,
+		})))
+		Expect(api.SortRules(makeSingleMatchRules([]api.Match{
+			singleHeaderMatch,
+			prefixMatch,
+		}))).To(Equal(makeRoutes([]api.Match{
+			prefixMatch,
+			singleHeaderMatch,
+		})))
+	})
+	It("is stable", func() {
+		Expect(api.SortRules(makeSingleMatchRules([]api.Match{
+			exactMatch,
+			otherExactMatch,
+		}))).To(Equal(makeRoutes([]api.Match{
+			exactMatch,
+			otherExactMatch,
+		})))
+		Expect(api.SortRules(makeSingleMatchRules([]api.Match{
+			otherExactMatch,
+			exactMatch,
+		}))).To(Equal(makeRoutes([]api.Match{
+			otherExactMatch,
+			exactMatch,
+		})))
+	})
+	It("handles rules with multiple matches", func() {
+		prefixMatchOnly := []api.Match{prefixMatch}
+		singleOrExact := []api.Match{singleHeaderMatch, exactMatch}
+		Expect(api.SortRules(makeMultiMatchRules([][]api.Match{
+			prefixMatchOnly,
+			singleOrExact,
+		}))).To(Equal([]api.Route{{
+			Match: exactMatch,
+			Hash:  api.HashMatches(singleOrExact),
+		}, {
+			Match: prefixMatch,
+			Hash:  api.HashMatches(prefixMatchOnly),
+		}, {
+			Match: singleHeaderMatch,
+			Hash:  api.HashMatches(singleOrExact),
+		}}))
```

Comment:
Might make sense to have a test with no rules or tests with exact match but different methods